scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/RuntimeReducerEstimator.scala

```diff
@@ -0,0 +1,107 @@
+package com.twitter.scalding.reducer_estimation
+
+import scala.collection.JavaConverters._
+import cascading.flow.FlowStep
+import cascading.tap.{ Tap, CompositeTap }
+import cascading.tap.hadoop.Hfs
+import org.apache.hadoop.mapred.JobConf
+import org.slf4j.LoggerFactory
+
+object RuntimeReducerEstimator {
+  val RuntimePerReducer = "scalding.reducer.estimator.runtime.per.reducer"
```

Comment:
Yep, runtime per reducer isn't a great name. "How long the reducers run" is along the lines of what I meant.

The history gives us a sequence of pairs (reducers, list of times for each reducer). `estimateTaskTime` takes each (reducers, list of times for each reducer) -> (reducers, time-required). Right now we've got an implementation of `estimateTaskTime` that uses the 50th percentile (median), and we could easily make one that uses the 80th percentile.

Using the pairs (num-reducers, time-required) to fit a function that predicts time-required from num-reducers is the interesting part.

Here was my approach (for the `BasicReducerEstimator`): 
I assumed that time-required = k / num-reducers for some constant k. So we calculate time-required * num-reducers for each point, and then use the median/mean of these to estimate k.

So the difference in our approaches is that I'm assuming an inverse relation, and you're assuming a line. I suppose that a linear fit would work if the data points are close enough together? Maybe we should take a better look at HRaven data to decide what kind of curve we want to fit.